File: Anushka-Pachauri/TwoPointer/ContainerWithMostWater.java
class ContainerWithMostWater {
    public int maxArea(int[] height) {
        int start=0;
        int end=height.length-1;
        int maxArea=0;
        while(start<end){
            if(height[end]>=height[start]){
                int area=height[start]*(end-start);
                maxArea=Math.max(maxArea,area);
                start++;
            }
            else{
                int area=height[end]*(end-start);
                maxArea=Math.max(maxArea,area);
                end--;
            }
        }
        return maxArea;
    }
}

File: NikhilKumarGupta/TwoPointer/container with most water.cpp
class Solution {
    public:
        int maxArea(vector<int>& height) {
            int n=height.size();
            int start=0;
            int end=n-1;
            int maxarea=INT_MIN;
            while(start<end){
                int width=end-start;
                int h=min(height[start],height[end]);
                int area=width*h;
                maxarea=max(maxarea,area);
                if(height[start]<height[end]){
                    start++;
                }
                else{
                end--;
                }
            }
    return maxarea;
        }
    };